lib/auth/join_circleci.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/api/types"
+	apiutils "github.com/gravitational/teleport/api/utils"
+	"github.com/gravitational/teleport/lib/circleci"
+	"github.com/gravitational/trace"
+)
+
+func (a *Server) checkCircleCIJoinRequest(ctx context.Context, req *types.RegisterUsingTokenRequest) error {
+	if req.IDToken == "" {
+		return trace.BadParameter("%q not provided for %q join request", "IDToken", types.JoinMethodCircleCI)
+	}
+	pt, err := a.GetToken(ctx, req.Token)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	token, ok := pt.(*types.ProvisionTokenV2)
+	if !ok {
+		return trace.BadParameter("%q join method only support ProvisionTokenV2", types.JoinMethodCircleCI)
```

Comment:
```suggestion
		return trace.BadParameter("%q join method only supports ProvisionTokenV2", types.JoinMethodCircleCI)
```

What do you think about including the actual type in the error message? Too much info?

Reply:
I can't see the harm in including it. As this error really will only ever be the result of programmer error, I can actually see it as being more helpful to include the type that has managed to sneak in. I'll add this.

Reply:
bf35db857b